### PR TITLE
Reset volume button in settings

### LINKED ISSUE #139: Add button to destroy volume from UI
We should add a button to the UI to destroy the docker volume and then reboot the application. This would allow users to "turn it off and on again" and solve any unforeseen database or project issues.

### PR DESCRIPTION
**Pull request by GrantPerkins**

### CODE REVIEW

**Comment by GrantPerkins** [comment]
Resolves #139

File: packages/client/src/index.tsx [hunk 1]
```diff
@@ -13,5 +13,5 @@ ReactDOM.render(
 
 // If you want your app to work offline and load faster, you can change
 // unregister() to register() below. Note this comes with some pitfalls.
-// Learn more about service workers: https://bit.ly/CRA-PWA
+// Learn more docs service workers: https://bit.ly/CRA-PWA
```
Comment (AustinShalit) [inline packages/client/src/index.tsx]: Revert

File: packages/client/src/serviceWorker.ts [hunk 2]
```diff
@@ -7,7 +7,7 @@
 // existing tabs open on the page have been closed, since previously cached
 // resources are updated in the background.
 
-// To learn more about the benefits of this project and instructions on how to
+// To learn more docs the benefits of this project and instructions on how to
```
Comment (AustinShalit) [inline packages/client/src/serviceWorker.ts]: Revert

File: packages/launcher/src/index.tsx [hunk 3]
```diff
@@ -12,5 +12,5 @@ ReactDOM.render(
 
 // If you want your app to work offline and load faster, you can change
 // unregister() to register() below. Note this comes with some pitfalls.
-// Learn more about service workers: https://bit.ly/CRA-PWA
+// Learn more docs service workers: https://bit.ly/CRA-PWA
```
Comment (AustinShalit) [inline packages/launcher/src/index.tsx]: Revert

File: packages/launcher/src/serviceWorker.ts [hunk 4]
```diff
@@ -7,7 +7,7 @@
 // existing tabs open on the page have been closed, since previously cached
 // resources are updated in the background.
 
-// To learn more about the benefits of this project and instructions on how to
+// To learn more docs the benefits of this project and instructions on how to
```
Comment (AustinShalit) [inline packages/launcher/src/serviceWorker.ts]: Revert

File: packages/server/src/resolvers/Mutation.ts [hunk 5]
```diff
@@ -4,6 +4,13 @@ export const Mutation: MutationResolvers = {
   resetDocker: async (parent, args, { docker }) => {
     return await docker.reset();
   },
+  resetVolume: async (parent, args, { docker, dataSources }) => {
+    return (
+      (await docker.resetVolume()) &&
+      (await dataSources.datasetService.reset()) &&
+      (await dataSources.projectService.reset())
```
Comment (AustinShalit) [inline packages/server/src/resolvers/Mutation.ts]: Do these need to have sequentially or can they happen in parallel?

**Review by AustinShalit — CHANGES_REQUESTED** [review]

File: packages/server/src/resolvers/Mutation.ts (hunk 5 above)
Comment (GrantPerkins) [reply packages/server/src/resolvers/Mutation.ts]: Now in parallel.

File: packages/server/src/resolvers/Mutation.ts [hunk 6]
```diff
@@ -4,6 +4,14 @@ export const Mutation: MutationResolvers = {
   resetDocker: async (parent, args, { docker }) => {
     return await docker.reset();
   },
+  resetVolume: async (parent, args, { docker, dataSources }) => {
+    const [volume, dataset, project] = await Promise.all([
+      docker.resetVolume(),
+      dataSources.datasetService.reset(),
+      dataSources.projectService.reset()
```
Comment (AustinShalit) [inline packages/server/src/resolvers/Mutation.ts]: I think you can use
```suggestion
      sequelize.sync({ force: true })
```
https://sequelize.org/master/class/lib/sequelize.js~Sequelize.html#instance-method-sync

to really reset the entire thing without leaving anything behind.

**Review by AustinShalit — CHANGES_REQUESTED** [review]

**Comment by AustinShalit** [comment]
Also, we should add a button/menu bar item in the launcher that'll really try to reset everything by calling docker delete.

**Comment by GrantPerkins** [comment]
image

**Review by AustinShalit — APPROVED** [review]

File: packages/server/src/resolvers/Mutation.ts [hunk 7]
```diff
@@ -1,9 +1,14 @@
 import { MutationResolvers } from "../schema/__generated__/graphql";
+import { sequelize } from "../store";
 
 export const Mutation: MutationResolvers = {
   resetDocker: async (parent, args, { docker }) => {
     return await docker.reset();
   },
+  resetVolume: async (parent, args, { docker, dataSources }) => {
```
Comment (AustinShalit) [inline packages/server/src/resolvers/Mutation.ts]: See automated review on this line: https://github.com/wpilibsuite/Axon/pull/216/checks?check_run_id=2943497452

**Review by AustinShalit — CHANGES_REQUESTED** [review]

**Review by AustinShalit — APPROVED** [review]

**Pull request merged**